File: src/view_controller/userList.ts
import { App } from "@slack/bolt";
import { postedReviewUserListModal } from "../view";
import { UserService } from "../service";

const userService = new UserService();

export default (app: App) => {
  /**
   * 「レビューを投稿したユーザ」ボタンが押下されたとき
   */
  app.action(
    "list_user_posted_review_action",
    async ({ ack, client, body }) => {
      await ack();

      if (body.type !== "block_actions") {
        return;
      }

      const users = await userService.fetchByPostedReview();

      if (!users) {
        throw new Error("Failed to retrieve users.");
      }

      await client.views.open({
        trigger_id: body.trigger_id,
        view: postedReviewUserListModal({
          callbackId: "review_modal",
          users,
        }),
      });
    }
  );
};

File: lambda/app/bee_slack_app/view_controller/user_list.py
from slack_bolt import App

from bee_slack_app.service.user import get_users_posted_review
from bee_slack_app.view import posted_review_user_list_modal


def user_list_controller(app: App) -> None:
    @app.action("list_user_posted_review_action")
    def list_user_posted_review_action(ack, body, client):
        """
        レビューを投稿したユーザモーダルを開く
        """
        ack()

        users = get_users_posted_review()

        client.views_open(
            trigger_id=body["trigger_id"],
            view=posted_review_user_list_modal(callback_id="review_modal", users=users),
        )
